src/plugins/discover/public/application/view_components/utils/use_search.ts

```diff
@@ -216,17 +221,30 @@ export const useSearch = (services: DiscoverViewServices) => {
           indexPattern?.title !== searchSource.getDataFrame()?.name
             ? searchSource.getDataFrame()?.name
             : indexPattern?.title,
+        queryStatus: {
+          status: rows.length > 0 ? ResultStatus.READY : ResultStatus.NO_RESULTS,
+          time: queryTime,
+        },
       });
     } catch (error) {
       // If the request was aborted then no need to surface this error in the UI
       if (error instanceof Error && error.name === 'AbortError') return;
 
+      let errorBody;
+      try {
+        errorBody = JSON.parse(error.body.message);
+      } catch (e) {
+        errorBody = error.body.message;
+      }
+
       data$.next({
-        status: ResultStatus.NO_RESULTS,
-        rows: [],
+        status: ResultStatus.ERROR,
+        queryStatus: {
+          status: ResultStatus.ERROR,
+          body: errorBody,
+          time: queryTime,
+        },
```

Comment:
```suggestion
        queryStatus: {
          status: ResultStatus.ERROR,
          body: errorBody,
          time: queryTime,
        },
```
Nit: Lets use just one status

Reply:
changed